From 87e253647795bc457b537fd5678954673442ae72 Mon Sep 17 00:00:00 2001
From: Vaclav Slavik <vslavik@gmail.com>
Date: Wed, 6 Feb 2013 20:08:41 +0100
Subject: [PATCH] Really fix compilation fix for wxWidgets 2.8.

The previous commit, 37a51b85a338bc2e24411e1d1caa62a8e2231bd8,
introduced another 2.8 incompatibility. Should be really fixed now.

Really fixes #513.
---
 src/catalog.cpp | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/catalog.cpp b/src/catalog.cpp
index ccb6c71..10b7ca6 100644
--- a/src/catalog.cpp
+++ b/src/catalog.cpp
@@ -1151,11 +1151,12 @@ inline bool CanEncodeStringToCharset(const wxString& s, wxMBConv& conv)
 {
     if (s.empty())
         return true;
+    const wxCharBuffer converted(s.mb_str(conv));
 #if wxCHECK_VERSION(2,9,0)
-    wxCharBuffer converted(s.mb_str(conv));
     if ( converted.length() == 0 )
 #else
-    if ( wxStrlen(s.mb_str(conv)) == 0 )
+    const char *cs = (const char*)converted;
+    if ( cs == NULL || strlen(cs) == 0 )
 #endif
         return false;
     return true;
-- 
1.8.1.6

